independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/InterceptorInfo.java

```diff
@@ -52,42 +53,78 @@
                 null, null, null, Collections.emptyList(), null, false);
         this.bindings = bindings;
         this.priority = priority;
-        List<MethodInfo> aroundInvokes = new ArrayList<>();
-        List<MethodInfo> aroundConstructs = new ArrayList<>();
-        List<MethodInfo> postConstructs = new ArrayList<>();
-        List<MethodInfo> preDestroys = new ArrayList<>();
+        aroundInvokes = new ArrayList<>();
+        aroundConstructs = new ArrayList<>();
+        postConstructs = new ArrayList<>();
+        preDestroys = new ArrayList<>();
 
         ClassInfo aClass = target.asClass();
-        while (aClass != null) {
-            for (MethodInfo method : aClass.methods()) {
-                if (Modifier.isStatic(method.flags())) {
-                    continue;
-                }
-                if (method.hasAnnotation(DotNames.AROUND_INVOKE)) {
-                    aroundInvokes.add(validateSignature(method));
+        processInheritedMethods(aClass, beanDeployment);
+
+        if (aroundConstructs.isEmpty() && aroundInvokes.isEmpty() && preDestroys.isEmpty() && postConstructs.isEmpty()) {
+            LOGGER.warnf("%s declares no around-invoke method nor a lifecycle callback!", this);
+        }
+    }
+
+    private void processInheritedMethods(ClassInfo aClass, BeanDeployment beanDeployment) {
+        DotName superTypeName = aClass.superName();
+        if (superTypeName != null && !DotNames.OBJECT.equals(superTypeName)) {
+            ClassInfo superClassInfo = getClassByName(beanDeployment.getBeanArchiveIndex(), superTypeName);
+            if (superClassInfo == null) {
+                throw new DefinitionException(
+                        "The interception super type " + superTypeName + " was not found in the index");
+            }
+            processInheritedMethods(superClassInfo, beanDeployment);
+        }
+
+        for (MethodInfo method : aClass.methods()) {
+            if (Modifier.isStatic(method.flags())) {
+                continue;
+            }
+            if (method.hasAnnotation(DotNames.AROUND_INVOKE)) {
+                //Map can not been used because order must be respected.
+                //remove overridden methods
+                for (MethodInfo info : aroundInvokes) {
+                    if (info.name().equals(method.name())) {
+                        aroundInvokes.remove(info);
```

Comment:
This is wrong actually, you're modifying a collection you're iterating. You have to use `Iterator.remove` (or iterate a copy of the collection).

Reply:
for this one, I know that s shitty, maybe a sorted hasmap with dedicated item with comparator. based on index is better. even for performance.
But I am not happy of the index technique so I want ot have this finished and discuss about solution design and refactoring.

Reply:
Sure. I was just curious about the current state -- plus the PR is not a draft anymore, so it's implicitly review-worthy :-)